src/orbitdb.js

```diff
@@ -23,98 +26,117 @@ AccessControllers.addAccessController({ AccessController: ModeratorAccessControl
-   const heads = await cache.getHeads()
-
-   const manifestHash = await cache.getManifest()
-   if (!manifestHash) return null
-   const manifest = await io.read(this._ipfs, manifestHash)
-   const dbType = manifest.type
-   const acAddress = manifest.accessController
-
-   // if no heads, quickly return with empty db
-   if (heads.length == 0) return dbType === 'feed' ? [] : {}
-
-   let acOpts
-   // TODO handle case better, could have spaces called threads
-   if (address.includes('thread') && !address.includes('_access') ) {
-     acOpts = { acAddress, readDB: this.readDB.bind(this)}
-   } else {
-     acOpts =  {skipManifest: true, type: 'legacy-ipfs-3box', write:[]}
-   }
-
-   const accessController = await AccessControllers.resolve({'_ipfs': this._ipfs}, acAddress, acOpts)
-
-   let log
-   for (const head of heads) {
-     const mergelog = await Log.fromEntryHash(this._ipfs, nullIdentity, head.hash, { logId: address, access: accessController, sortFn: undefined, length: amount})
-     if (log) {
-       await log.join(mergelog)
-     } else {
-       log = mergelog
-     }
-   }
-
-   // Get overlay db type to build db index
-   if (!dbTypeIndex[dbType]) throw new Error('Type not supported')
-   const index = new dbTypeIndex[dbType]()
-   index.updateIndex(log)
-
-   // Map db index to typical return all form
-   if (dbType === 'feed') {
-     const entry = k => index._index[k]
-     const mapFunc = threadMetaData ? k => entry(k) : k => entry(k).payload.value
-     return Object.keys(index._index).map(mapFunc)
-   }
-
-   return index._index
- }
-
- async getThreadAddress (name, moderator, members) {
-   const modAC = moderatorAC(moderator, members)
-   const dbModName = `${name}/_access`
-   const acDBAddress = await this.dbAddress (modAC, dbModName, 'feed')
-
-   const ac = threadAC (name, moderator, members, acDBAddress)
-   return this.dbAddress(ac, name, 'feed')
- }
-
- async dbAddress (accessController, dbName, dbType, format) {
-   const accessControllerAddress = await AccessControllers.create({'_ipfs': this._ipfs}, accessController.type, accessController)
-   const manifestHash = await createDBManifest(this._ipfs, dbName, dbType, accessControllerAddress, {format})
-   return path.join('/orbitdb', manifestHash, dbName)
- }
+    const cache = await this._orbitCache.load(address)
+    const heads = await cache.getHeads()
+
+    const manifestHash = await cache.getManifest()
+    if (!manifestHash) return null
+    const manifest = await io.read(this._ipfs, manifestHash)
+    const dbType = manifest.type
+    const acAddress = manifest.accessController
+
+    // if no heads, quickly return with empty db
+    if (heads.length == 0) return dbType === 'feed' ? [] : {}
+
+    let acOpts
+    // TODO handle case better, could have spaces called threads
+    if (address.includes('thread') && !address.includes('_access') ) {
+      acOpts = {
+        acAddress,
+        readDB: this.readDB.bind(this)
```

Comment:
Is the resolver not needed here?

Reply:
nope, only used by the legacy access controller to get the did's public key - otherwise the resolver is invoked by the identity provider itself

Reply:
Ok, thanks for clarifying 👍